torch/_tensor.py

```diff
@@ -27,16 +28,20 @@
 )
 
 
-def _handle_torch_function_and_wrap_type_error_to_not_implemented(f):
-    assigned = functools.WRAPPER_ASSIGNMENTS
+_P = ParamSpec("_P")
+_TensorLike = TypeVar("_TensorLike", bound=_C.TensorBase)
 
-    @functools.wraps(f, assigned=assigned)
-    def wrapped(*args, **kwargs):
+
+def _handle_torch_function_and_wrap_type_error_to_not_implemented(
+    f: Callable[Concatenate[_TensorLike, _P], "Tensor"],
+) -> Callable[Concatenate[_TensorLike, _P], "Tensor"]:
+    @functools.wraps(f)
+    def wrapped(self: _TensorLike, *args: _P.args, **kwargs: _P.kwargs) -> "Tensor":
         try:
-            # See https://github.com/pytorch/pytorch/issues/75462
```

Comment:
This issue is still open? Was it really resolved by this PR? https://github.com/pytorch/pytorch/issues/75462

Reply:
This commit _mostly_ resolves this issue. 😁 

This pull request here is all the fixes that don't change the behavior of `Tensor`.

There's a PR stacked on top of it, https://github.com/pytorch/pytorch/pull/146894, which resolves the remaining handful of cases.

I split that one out because it changes the observed behavior of `Tensor` slightly - it allows Python `bool`s and `int`s to appear on the left hand side of the bitwise operators `&`, `|` and `^`, and there were a few tests ensuring this didn't happen.

It's not just for neatness' sake, but that I think the behavior changing PR is more likely to get reverted upstream by someone even if it passes all the CI.